### DISCUSSION TITLE
Unable to run rabbitmq-server on Centos 9 Stream

### DISCUSSION THREAD

**Discussion by SubZero77** [rabbitmq/rabbitmq-server#6439] [Questions] [answered]
I had a perfectly working server on Centos 9 Stream with Nextcloud 25.0.1, Nginx 1.23.2, Postgresql server 15 but rabbitmq-server does not start on it. It is installed rabbitmq-server-3.11.3-1.8 and erlang-25.1.1-1.el8.
Here is the startup log:
[root@nc /]# systemctl start rabbitmq-server.service
Job for rabbitmq-server.service failed because the control process exited with error code.
See "systemctl status rabbitmq-server.service" and "journalctl -xeu rabbitmq-server.service" for details.
[root@nc /]# journalctl -xeu rabbitmq-server.service
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.815712+03:00 [error] <0.131.0>                {restart_type,transient},
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.815712+03:00 [error] <0.131.0>                {significant,false},
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.815712+03:00 [error] <0.131.0>                {shutdown,5000},
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.815712+03:00 [error] <0.131.0>                {child_type,worker}]
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.815712+03:00 [error] <0.131.0>
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.816454+03:00 [error] <0.129.0>   crasher:
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.816454+03:00 [error] <0.129.0>     initial call: application_master:init/4
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.816454+03:00 [error] <0.129.0>     pid: <0.129.0>
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.816454+03:00 [error] <0.129.0>     registered_name: []
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.816454+03:00 [error] <0.129.0>     exception exit: {{shutdown,
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.816454+03:00 [error] <0.129.0>                          {failed_to_start_child,prelaunch,
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.816454+03:00 [error] <0.129.0>                              {epmd_error,"nc",timeout}}},
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.816454+03:00 [error] <0.129.0>                      {rabbit_prelaunch_app,start,[normal,[]]}}
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.816454+03:00 [error] <0.129.0>       in function  application_master:init/4 (application_master.erl, line 142)
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.816454+03:00 [error] <0.129.0>     ancestors: [<0.128.0>]
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.816454+03:00 [error] <0.129.0>     message_queue_len: 1
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.816454+03:00 [error] <0.129.0>     messages: [{'EXIT',<0.130.0>,normal}]
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.816454+03:00 [error] <0.129.0>     links: [<0.128.0>,<0.44.0>]
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.816454+03:00 [error] <0.129.0>     dictionary: []
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.816454+03:00 [error] <0.129.0>     trap_exit: true
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.816454+03:00 [error] <0.129.0>     status: running
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.816454+03:00 [error] <0.129.0>     heap_size: 376
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.816454+03:00 [error] <0.129.0>     stack_size: 28
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.816454+03:00 [error] <0.129.0>     reductions: 165
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.816454+03:00 [error] <0.129.0>   neighbours:
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.816454+03:00 [error] <0.129.0>
ноя 21 20:30:20 nc.server.com rabbitmq-server[13076]: 2022-11-21 20:30:20.821561+03:00 [notice] <0.44.0> Application rabbitmq_prelaunch exited with reason: {{shutdown,{failed_to_start_child,prelaunch,{epmd_error,"nc",timeout}}},{rabbit_prelaunch_app,start,[normal,[]]}}
ноя 21 20:30:22 nc.server.com rabbitmq-server[13076]: {"Kernel pid terminated",application_controller,"{application_start_failure,rabbitmq_prelaunch,{{shutdown,{failed_to_start_child,prelaunch,{epmd_error,\"nc\",timeout}}},{rabbit_prelaunch_app,start,[normal,[]]}}}"}
ноя 21 20:30:22 nc.server.com rabbitmq-server[13076]: Kernel pid terminated (application_controller) ({application_start_failure,rabbitmq_prelaunch,{{shutdown,{failed_to_start_child,prelaunch,{epmd_error,"nc",timeout}}},{rabbit_prelaunch_app,start,[normal,[]]}}})
ноя 21 20:30:22 nc.server.com rabbitmq-server[13076]:
ноя 21 20:30:22 nc.server.com rabbitmq-server[13076]: Crash dump is being written to: erl_crash.dump...done
ноя 21 20:30:22 nc.server.com systemd[1]: rabbitmq-server.service: Main process exited, code=exited, status=1/FAILURE
░░ Subject: Unit process exited
░░ Defined-By: systemd
░░ Support: https://access.redhat.com/support
░░
░░ An ExecStart= process belonging to unit rabbitmq-server.service has exited.
░░
░░ The process' exit code is 'exited' and its exit status is 1.
ноя 21 20:30:22 nc.server.com systemd[1]: rabbitmq-server.service: Failed with result 'exit-code'.
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: https://access.redhat.com/support
░░
░░ The unit rabbitmq-server.service has entered the 'failed' state with result 'exit-code'.

**Comment by SubZero77** [comment, marked as answer]
I found the solution myself. After running my own script adding ip fqdn short name in /etc/hosts:
sed -i '/^[^#]*::1/s/^/#/' /etc/hosts && addr="$(hostname -I)$HOSTNAME ${HOSTNAME%%.*}" && sed -zi '/'"$addr"'/!s/$/'"$addr"'\n/' /etc/hosts
everything worked as it should. It is very strange that no other program or service needed this :)

**Reply by michaelklishin** [reply]
RabbitMQ (or rather, the Erlang runtime) can use short or long (FQ) host names but since they are part of [node identifiers](https://www.rabbitmq.com/clustering.html#node-names), all nodes must be able to resolve their own hostnames, must use the same hostname type, and are sensitive to hostname changes.
